src/controllers/simplified-account/settings/card-payments/apple-pay.controller.js

```diff
@@ -0,0 +1,30 @@
+const { response } = require('@utils/response')
+const { formatSimplifiedAccountPathsFor } = require('@utils/simplified-account/format')
+const { onOrOffToBool } = require('@utils/on-or-off')
+const paths = require('@root/paths')
+const { toggleApplePay } = require('@services/card-payments.service')
+const flashSuccess = require('@utils/flash-success')
+
+function get (req, res) {
+  const url = req.originalUrl.split('?')[0]
+  const applePay = req.account?.allowApplePay
+  response(req, res, 'simplified-account/settings/card-payments/apple-pay', {
+    formAction: url,
+    currentState: applePay ? 'on' : 'off',
+    backLink: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.cardPayments.index, req.service.externalId, req.account.type)
+  })
+}
+
+async function post (req, res) {
+  const userPreference = onOrOffToBool(req.body.applePay)
+  const serviceExternalId = req.service.externalId
+  const gatewayAccountId = req.account.id
+  await toggleApplePay(gatewayAccountId, userPreference)
+  flashSuccess(req, `Apple Pay successfully ${userPreference ? 'enabled' : 'disabled'}`)
```

Comment:
The designs don't include success banners when updates are made, since the user will see the change on the index page perhaps these aren't necessary - has this been reviewed by the design team?

Reply:
Removed.